contracts/NodeOperatorRegistry.sol

```diff
@@ -0,0 +1,642 @@
+
+        require(
+            no.status == Operator.NodeOperatorStatus.EXIT,
+            "Operator status not EXIT"
+        );
+
+        require(_proof.length != 0, "Empty proof");
+        require(_accumFeeAmount != 0, "AccumFeeAmount is ZERO");
+        require(_index != 0, "index is ZERO");
+
+        IValidator(no.validatorContract).claimFee(
+            _accumFeeAmount,
+            _index,
+            _proof
+        );
+
+        emit ClaimFee(
+            operatorId,
+            no.validatorId,
+            _accumFeeAmount,
+            _index,
+            _proof
+        );
+    }
+
+    /// @notice Allows to update commission rate
+    /// @param _newCommissionRate new commission rate
+    function updateOperatorCommissionRate(
+        uint256 _operatorId,
+        uint256 _newCommissionRate
+    ) public override whenNotPaused userHasRole(UPDATE_COMMISSION_RATE_ROLE) {
+        Operator.NodeOperator memory no = operators[_operatorId];
+        if (no.status != Operator.NodeOperatorStatus.STAKED)
+            revert("Operator status no STAKED");
+
+        IValidator(no.validatorContract).updateCommissionRate(
+            no.validatorId,
+            _newCommissionRate
+        );
+
+        no.commissionRate = state.commissionRate;
+
+        emit UpdateCommissionRate(no.validatorId, _newCommissionRate);
+    }
+
+    /// @notice Allows to unjail the validator and turn his status from UNSTAKED to STAKED.
+    function unjail() external override whenNotPaused {
+        require(state.unjail, "Unjail is disabled");
+        
+        uint256 operatorId = operatorOwners[msg.sender];
+        require(operatorId != 0, "Operator not exists");
+
+        Operator.NodeOperator memory no = operators[operatorId];
+
+        require(
+            no.status == Operator.NodeOperatorStatus.UNSTAKED,
+            "Operator status not UNSTAKED"
+        );
+
+        IValidator(no.validatorContract).unjail(no.validatorId);
+
+        no.status = Operator.NodeOperatorStatus.STAKED;
+        state.totalStakedNodeOpearator++;
+        state.totalUnstakedNodeOpearator--;
+
+        emit Unjail(operatorId, no.validatorId);
+    }
+
+    /// @notice Allows to update the stake amount and heimdall fees
+    /// @param _minAmountStake min amount to stake
+    /// @param _maxAmountStake max amount to stake
+    /// @param _minHeimdallFees min amount of heimdall fees
+    /// @param _maxHeimdallFees max amount of heimdall fees
+    function setStakeAmountAndFees(
+        uint256 _minAmountStake,
+        uint256 _maxAmountStake,
+        uint256 _minHeimdallFees,
+        uint256 _maxHeimdallFees
+    ) public userHasRole(UPDATE_STAKE_HEIMDALL_FEES_ROLE) {
+        require(
+            _minAmountStake > 0 && _minAmountStake <= _maxAmountStake,
+            "Invalid amount"
+        );
+        require(
+            _minHeimdallFees > 0 && _minHeimdallFees <= _maxHeimdallFees,
+            "Invalid heimdallFees"
+        );
```

Comment:
I see that this require is used at multiple places. Maybe you could make a modifier out of it just to improve readibility. Same goes for the line above.